app/src/main/java/by/dutov/jee/service/Finance.java

```diff
@@ -0,0 +1,128 @@
+package by.dutov.jee.service;
+
+import by.dutov.jee.people.Person;
+import by.dutov.jee.people.Teacher;
+import by.dutov.jee.repository.RepositoryFactory;
+import by.dutov.jee.repository.person.PersonDAO;
+import by.dutov.jee.utils.CommandServletUtils;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Slf4j
+public class Finance {
+    private static final Map<Teacher, Map<Integer, Double>> salaryHistory;
+    private static final int CURRENT_MONTH = LocalDate.now().getMonthValue();
+    private static final PersonDAO<Person> DAO_REPOSITORY = RepositoryFactory.getDaoRepository();
+    private static volatile Finance instance;
+    private static final String MIN_RANGE = "minRange";
+    private static final String MAX_RANGE = "maxRange";
+
+    public Finance() {
+        //singleton
+    }
+
+    public Map<Teacher, Map<Integer, Double>> getSalaryHistory() {
+        return salaryHistory;
+    }
+
+    public static Finance getInstance() {
+        if (instance == null) {
+            synchronized (Finance.class) {
+                if (instance == null) {
+                    instance = new Finance();
+                }
+            }
+        }
+        return instance;
+    }
+
+    static {
+        salaryHistory = new HashMap<>();
+        final Optional<? extends Person> teacher = DAO_REPOSITORY.find("teacher");
+        final Optional<? extends Person> teacher1 = DAO_REPOSITORY.find("teacher1");
+        if (teacher.isPresent() && teacher1.isPresent()) {
+            for (int i = 1; i < 11; i++) {
+                Finance.getInstance().saveSalary((Teacher) teacher.get(), i, i * 110.0);
+                Finance.getInstance().saveSalary((Teacher) teacher1.get(), i, i * 100.0);
+            }
+            Finance.getInstance().saveSalary((Teacher) teacher.get(), CURRENT_MONTH, ((Teacher) teacher.get()).getSalary());
+            Finance.getInstance().saveSalary((Teacher) teacher1.get(), CURRENT_MONTH, ((Teacher) teacher1.get()).getSalary());
+        }
+    }
+
+    public double averageSalary(int minRange, int maxRange, Teacher teacher) {
+        int rangeCount = maxRange - minRange + 1;
+        double avgSal = 0.0;
+        if (minRange >= 1 && maxRange <= CURRENT_MONTH
+                && rangeCount > 0) {
```

Comment:
А лучше не строить лесенку, а сделать 
```suggestion
        if (!(minRange >= 1 && maxRange <= CURRENT_MONTH
                && rangeCount > 0)) {
                return -1;
         }       
```
А дальше уже в том же уровне продолжать проверки.